nixos/modules/tasks/swraid.nix

```diff
@@ -59,12 +67,10 @@ in {
         $out/bin/mdadm --version
       '';
 
-      extraFiles."/etc/mdadm.conf".source = pkgs.writeText "mdadm.conf" config.boot.swraid.mdadmConf;
+      extraFiles."/etc/mdadm.conf" = mdadm_conf;
```

Comment:
maybe because of this. `extraFiles."/etc/mdadm.conf".source = mdadm_conf.source;` could fix it.

Reply:
It does. However, I can't figure out why and I would like to adjust the test so that it doesn't happen again, because we explicitly thought this would work and the tests and my live system agreed with me.

Even when I explicitly set `environment.etc."mdadm.conf".enable = true` the test doesn't break and I also do not even see why it should, as nothing iterates over the superfluous attributes.